tests/Aspire.Hosting.Tests/ResourceNotificationTests.cs

```diff
@@ -291,6 +293,87 @@ public async Task WaitingOnResourceThrowsOperationCanceledExceptionIfResourceDoe
         });
     }
 
+    [Fact]
+    public async Task PublishLogsStateTextChangesCorrectly()
+    {
+        var resource1 = new CustomResource("resource1");
+        var logger = new FakeLogger<ResourceNotificationService>();
+        var notificationService = new ResourceNotificationService(logger, new TestHostApplicationLifetime());
+
+        await notificationService.PublishUpdateAsync(resource1, snapshot => snapshot with { State = "SomeState" });
+
+        var logs = logger.Collector.GetSnapshot();
+
+        // Initial state text, log just the new state
+        Assert.Contains(logs, l => l.Level == LogLevel.Debug && l.Message.Contains("Resource resource1/resource1 changed state: SomeState"));
```

Comment:
Do we want to assert how many things are in the logs?

Reply:
Yeah good point. I don't like making log tests perfectly tied to the logs as it makes them super fragile but in this case asserting the number of logs makes sense.

Reply:
Adding assertion for the log counts.